integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/TopicBasicsTestCase.java

```diff
@@ -0,0 +1,159 @@
+            Assert.assertNotNull(subscriber2.receiveMessage(ClientConstants.CLIENT_WAIT_DURATION),
+                    "Topic not received by the subscriber2");
+
+            //send 500 messages
+            sender.sendMessages(500, "test message");
+            Assert.assertEquals(subscriber.getMessageCount(), 500, "Did not receive all the topics by "
+                    + "subscriber1");
+            Assert.assertEquals(subscriber2.getMessageCount(), 500, "Did not receive all the topics by "
+                    + "subscriber2");
+        } finally {
+            if (sender != null) {
+                sender.closeSender();
+                subscriber.closeReceiver();
+            }
+        }
+    }
+
+    /**
+     * Test for durable subscription.
+     * 1. Create two durable subscribers.
+     * 2. Send 10 messages and check for their reception.
+     * 3. Close one subscriber.
+     * 4. Send 10 messages.
+     * 5. Start the closed subscriber again.
+     * 6. Check the reception of messages.
+     * @throws JMSException
+     * @throws NamingException
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    @Test(groups = "wso2.mb.topic")
+    public void testDurableSubscription() throws JMSException, NamingException, IOException, InterruptedException {
+        TopicSender sender = null;
+        TopicReceiver subscriber = null;
+        TopicReceiver subscriber2 = null;
+        try {
+            subscriber = new TopicReceiver("testTopic2", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE, configurationReader,
+                    "sub1");
+
+            subscriber2 = new TopicReceiver("testTopic2", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE, configurationReader,
+                    "sub2");
+
+
+            //send 10 messages
+            sender = new TopicSender("testTopic2", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE, configurationReader);
+            sender.sendMessages(10, "test message");
+
+            Assert.assertEquals(subscriber.getMessageCount(), 10, "All the published topics not received by "
+                    + "subscriber");
+            Assert.assertEquals(subscriber2.getMessageCount(), 10, "All the published topics not received by "
+                    + "subscriber2");
+
+            //disconnect subscriber2
+            subscriber2.closeReceiver();
+
+            //send another 10 messages
+            sender.sendMessages(10, "test message");
+            subscriber2 = new TopicReceiver("testTopic2", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE, configurationReader,
+                    "sub2");
+            Assert.assertEquals(subscriber.getMessageCount(), 10, "All the published topics not received");
+            Assert.assertEquals(subscriber2.getMessageCount(), 10, "All the published topics not received by "
+                    + "subscriber2 after reconnection");
+        } finally {
+            if (sender != null) {
+                sender.closeSender();
+                subscriber.closeReceiver();
```

Comment:
Same as above. Do we need to do closeReceiver only is sender is not null?

Reply:
fixed.